### PR TITLE
[TemplateAdapter.php] Rewrote hardcoded SQL to query object

### PR DESCRIPTION
**Pull request by pe7er**
Rewrote 2 SQL queries to query object format.
This code replaces PR #12748 "JInstallerAdapterTemplate - rewrote hardcoded SQL to object"

### CODE REVIEW

File: libraries/src/Installer/Adapter/TemplateAdapter.php [hunk 1]
```diff
@@ -456,7 +456,11 @@ public function uninstall($id)
 
 		// Deny remove default template
 		$db = $this->parent->getDbo();
-		$query = "SELECT COUNT(*) FROM #__template_styles WHERE home = '1' AND template = " . $db->quote($name);
+		$query = $db->getQuery(true)
+			->select('COUNT(*)')
+			->from($db->qn('#__template_styles'))
+			->where($db->qn('home') . ' = 1')
```
Comment (alikon) [inline libraries/src/Installer/Adapter/TemplateAdapter.php]: same as #17921 
the `home` field is declared as char/varchar in the` #__template_styles` table so should be quoted as a string

**Review by alikon — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
@alikon are this PR and #17921 the same and on should be closed?

**Comment by pe7er** [comment]
@franz-wohlkoenig It looks similar like #17921 but it's a different file.

**Comment by pe7er** [comment]
@alikon Thanks your correction!
I've changed the SQL.

**Comment by pe7er** [comment]
Changed 
->where($db->qn('home') . ' = \'1\' ')
to
->where($db->qn('home') . ' = ' . $db->q('1'))

**Review by alikon — APPROVED** [review]

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 7268bdfb23c3742062403181d669c1587736c4d4<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17923">issues.joomla.org/tracker/joomla-cms/17923</a>.</sub>

File: libraries/src/Installer/Adapter/TemplateAdapter.php [hunk 2]
```diff
@@ -512,12 +516,15 @@ public function uninstall($id)
...
-
+		$subQuery = $db->getQuery(true)
+			->select('s.id')
+			->from('#__template_styles s')
+			->where($db->qn('s.template') . ' = ' . $db->q(strtolower($name)))
+			->where($db->qn('s.client_id') . ' = ' . $db->q($clientId));
+		$query->clear()
+			->update($db->qn('#__menu'))
+			->set($db->qn('template_style_id') . ' = 0')
+			->where($db->qn('template_style_id') . ' in (' . (string) $subQuery . ')');
```
Comment (Quy) [inline libraries/src/Installer/Adapter/TemplateAdapter.php]: Capitalize `in`

File: libraries/src/Installer/Adapter/TemplateAdapter.php [hunk 3]
```diff
@@ -512,12 +516,15 @@ public function uninstall($id)
...
 		// Set menu that assigned to the template back to default template
-		$query = 'UPDATE #__menu'
-			. ' SET template_style_id = 0'
-			. ' WHERE template_style_id in ('
-			. '	SELECT s.id FROM #__template_styles s'
-			. ' WHERE s.template = ' . $db->quote(strtolower($name)) . ' AND s.client_id = ' . $clientId . ')';
-
+		$subQuery = $db->getQuery(true)
+			->select('s.id')
+			->from('#__template_styles s')
```
Comment (Quy) [inline libraries/src/Installer/Adapter/TemplateAdapter.php]: Change to `->from($db->qn('#__template_styles', 's'))`

File: libraries/src/Installer/Adapter/TemplateAdapter.php [hunk 4]
```diff
@@ -512,12 +516,15 @@ public function uninstall($id)
...
+			->where($db->qn('s.template') . ' = ' . $db->q(strtolower($name)))
+			->where($db->qn('s.client_id') . ' = ' . $db->q($clientId));
```
Comment (Quy) [inline libraries/src/Installer/Adapter/TemplateAdapter.php]: Remove `$db->q`

**Comment by ghost** [comment]
@alikon can you please describe how you have tested?

**Comment by alikon** [comment]
install templates

**Comment by ghost** [comment]
@alikon install any Template or Core-Template?

**Comment by alikon** [comment]
any should be enough

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 7268bdfb23c3742062403181d669c1587736c4d4<br /><br />Installation of "Free Joomla 3.5 Template – jPeople" (http://www.freshjoomlatemplates.com/free-joomla-3-5-template-jpeople/) works.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17923">issues.joomla.org/tracker/joomla-cms/17923</a>.</sub>

**Comment by ghost** [comment]
RTC after two successful tests.

**Comment by mbabker** [comment]
Has some requested changes that should be implemented before merging.

**Comment by brianteeman** [comment]
@pe7er could you make the requested changes please so that this can be merged

**Comment by alikon** [comment]
@pe7er  i've made a pr https://github.com/pe7er/joomla-cms/pull/21 to your branch with the updated requested by @Quy

**Comment by pe7er** [comment]
Thanks @alikon! 
I've merged your PR into this one

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 1169ef4c32bf91dcc4770d9e3022f61d51e0f358<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17923">issues.joomla.org/tracker/joomla-cms/17923</a>.</sub>

**Comment by alikon** [comment]
can someone remove the RTC, update requested status, and  test

**Comment by ghost** [comment]
@alikon set Status back on "Pending" at Issue Tracker.

**Comment by alikon** [comment]
thanks folks, just another one little effort, please test

**Comment by fancyFranci** [comment]
I have tested this item :white_check_mark: successfully on 1169ef4c32bf91dcc4770d9e3022f61d51e0f358<br /><br />**Uninstalling** the template was successful.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17923">issues.joomla.org/tracker/joomla-cms/17923</a>.</sub>

**Comment by Quy** [comment]
@brianteeman Please remove `Updates Requested` label. Thanks.

**Comment by brianteeman** [comment]
@quy removed - lets see if the bot puts them back

**Pull request merged**